executor/insert_common.go

```diff
@@ -470,6 +471,45 @@ func insertRowsFromSelect(ctx context.Context, base insertCommon) error {
 	memUsageOfRows := int64(0)
 	memUsageOfExtraCols := int64(0)
 	memTracker := e.memTracker
+
+	var (
+		dataBatch [][]types.Datum
+		dataCh    chan [][]types.Datum
+	)
+	etlConcurrency, etlBatchSize := sessVars.ETLConcurrency, sessVars.ETLBatchSize
+	if etlMode && etlConcurrency > 0 {
+		workers := createFromSelectWorkers(ctx, base, etlConcurrency)
+		if len(workers) == 0 {
+			// fallback to single thread.
+			etlMode = false
```

Comment:
How about reporting an error here as we expect that the concurrency should work, and we don't need to run it in sequential mode.

Reply:
Report error when build session failed.